backend/app/services/session.py

```diff
@@ -46,3 +48,23 @@ async def delete_session(db_session: AsyncSession, session_id: str, current_user
 
     await db_session.delete(session_obj)
     await db_session.commit()
+
+async def get_session_ranking(db_session: AsyncSession, session_id: str) -> list[UserSessionRankingOut]:
+    session = await get_session_by_id(db_session, session_id)
+    participants = session.participants
+
+    if not participants:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="No participants found for this session")
```

Comment:
I'm not sure about it. Returning an empty list will be better, because the frontend may want to use this endpoint when any student hasn't joined.

Reply:
Agree